how-to/workspace-platform-starter/client/src/framework/platform/interopbroker.ts

```diff
@@ -461,26 +499,101 @@ export function interopOverride(
 				fdc3OpenOptions.context
 			);
 			try {
+				const isOpenByIntent = this._openOptions?.openStrategy === "INTENT";
+				let appId: string | undefined;
+				let instanceId: string | undefined;
+
 				const requestedApp = await getApp(requestedId);
 				if (isEmpty(requestedApp)) {
 					throw new Error(OpenError.AppNotFound);
 				}
-				const result = await this.launchAppWithIntent(requestedApp, openAppIntent);
-				let appId: string;
-				let instanceId: string | undefined;
-				if (typeof result.source === "string") {
-					appId = result.source;
+
+				if (isOpenByIntent) {
+					const result = await this.launchAppWithIntent(requestedApp, openAppIntent);
+					if (typeof result.source === "string") {
+						appId = result.source;
+					} else {
+						appId = result.source.appId;
+						instanceId = result.source.instanceId;
+					}
 				} else {
-					appId = result.source.appId;
-					instanceId = result.source.instanceId;
+					const platformIdentities = await launch(requestedApp);
+					if (!isEmpty(platformIdentities) && platformIdentities?.length > 0) {
+						appId = platformIdentities[0].appId;
+						const openTimeout: number | undefined = this._openOptions?.connectionTimeout;
+						// if we have a snapshot and multiple identities we will not wait as not all of them might not support intents.
+						instanceId = await this.onConnectionClientReady(platformIdentities[0], openTimeout);
+						if (platformIdentities.length > 1) {
+							logger.warn(
+								"Open can only return one app and instance id and multiple instances were launched as a result. Returning the first instance. Returned instances: ",
+								platformIdentities
+							);
+						}
+						if (!isEmpty(fdc3OpenOptions?.context)) {
+							const contextTimeout: number | undefined = this._intentOptions?.intentTimeout;
+							const contextTypeName = fdc3OpenOptions.context.type;
+							// if we have a snapshot and multiple identities we will not wait as not all of them might not support intents.
+							const clientReadyInstanceId = await this.onContextClientReady(
+								platformIdentities[0],
+								contextTypeName,
+								contextTimeout
+							);
+
+							let trackedHandler: ContextRegistrationEntry | undefined;
+
+							const trackedContextHandlers = this._trackedContextHandlers[contextTypeName];
+							if (Array.isArray(trackedContextHandlers)) {
+								const trackedContextHandler = trackedContextHandlers.find(
```

Comment:
You could combine these without the if, doesn't matter if its undefined i.e. trackedHandler = trackedContextHandlers.find

Reply:
undefined.find throws an error. if there is no context handler for e.g. mytype it will return undefined